File: leetcode26.c
#include <stdio.h>

int removeDuplicates(int* nums, int numsSize){
    if (numsSize <= 1)
    {
        return numsSize;
    }
    
    int offset = 0;
    for (int i = 1; i < numsSize; i++) {
        if (nums[i] != nums[offset]) {
            offset++;
            nums[offset] = nums[i];
        }
    }

    return offset + 1;
}

int main() {
    int list[10] = {0,0,1,1,1,2,2,3,3,4};
    // int list[3] = {0,0,1};
    int len = removeDuplicates(list, 10);
    printf("len ==> %d\n", len);
    for (int i = 0; i < 10; i++)
    {
        printf("%d", list[i]);
    }
    
    return 0;
}

File: leetcode26.py
class Solution:
    def removeDuplicates(self, nums):
        sameCount = 0
        preNum = None
        totalLength = len(nums)
        for num in nums:
            index =  nums.index(num)
            print(index)
            if totalLength - sameCount <= index:
                break

            if preNum == num:
                samCount += 1
                temp = nums[index]
                nums[index] = nums[totalLength-sameCount]
                nums[totalLength-sameCount] = temp
        
        print(nums)
                

test = Solution().removeDuplicates([1, 1, 2])